app/src/test/java/com/android/sample/model/image/ImageRepositoryFirestoreTest.kt

```diff
@@ -0,0 +1,207 @@
+package com.android.sample.model.image
+
+import android.graphics.Bitmap
+import android.net.Uri
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.setValue
+import com.google.android.gms.tasks.OnSuccessListener
+import com.google.android.gms.tasks.Tasks
+import com.google.firebase.firestore.CollectionReference
+import com.google.firebase.firestore.DocumentReference
+import com.google.firebase.firestore.DocumentSnapshot
+import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.storage.FirebaseStorage
+import com.google.firebase.storage.ListResult
+import com.google.firebase.storage.StorageReference
+import com.google.firebase.storage.UploadTask
+import java.io.ByteArrayOutputStream
+import junit.framework.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.mockito.ArgumentMatchers.anyString
+import org.mockito.Mock
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.`when`
+import org.mockito.MockitoAnnotations
+import org.mockito.kotlin.any
+
+class ImageRepositoryFirestoreTest {
+
+  @Mock private lateinit var mockFirestore: FirebaseFirestore
+  @Mock private lateinit var mockStorage: FirebaseStorage
+  @Mock private lateinit var mockStorageRef: StorageReference
+  @Mock private lateinit var mockUploadTask: UploadTask
+  private lateinit var imageRepository: ImageRepositoryFirestore
+  @Mock private lateinit var mockDocumentSnapshot: DocumentSnapshot
+  @Mock private lateinit var mockDocumentReference: DocumentReference
+  @Mock private lateinit var mockCollectionReference: CollectionReference
+  @Mock private lateinit var mockListResult: ListResult
+  @Mock private lateinit var bitmapHelper: BitmapHelper
+
+  @Before
+  fun setUp() {
+    MockitoAnnotations.openMocks(this)
+    `when`(mockStorage.reference).thenReturn(mockStorageRef)
+    `when`(mockFirestore.collection("users")).thenReturn(mockCollectionReference)
+    `when`(mockCollectionReference.document(anyString())).thenReturn(mockDocumentReference)
+    imageRepository = ImageRepositoryFirestore(mockFirestore, mockStorage)
+  }
+
```

Comment:
Tests the successful uploading of a profile picture, including compressing the image, uploading to Firebase Storage, and updating the Firestore document with the image URL.